src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php

```diff
@@ -82,7 +84,23 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ])
             ->add('sendemail', SwitchType::class, [
                 'data' => true,
-            ]);
+            ])
+        ;
+
+        $builder->get('entity')
+            ->addModelTransformer(new CallbackTransformer(
+                function ($entity) {
+                    if (null === $entity) {
+                        return $entity;
+                    }
+
+                    return is_numeric($entity) ? $entity : Entity::getFromName($entity);
```

Comment:
this sounds weird, can you explain what is done here?

Reply:
$entity can be both string (for example when you click "Export" in products page, it adds ?import_type=products query param, so it converts this string to actual value which is integer) and integer, which is actual import entity value.